src/pages/index.js

```diff
@@ -82,39 +118,82 @@ function renderCard(cardData) {
 
 function handleAddCardFormSubmit(inputData) {
   const { card__title, card__url } = inputData;
-  renderCard({ name: card__title, link: card__url });
-  addCardFormValidator.disableSubmitButton();
-  newCardPopup.close();
-  addNewCardForm.reset();
+  newCardPopup.setLoading(true);
+  api
+    .createCard({ name: card__title, link: card__url })
+    .then((newCard) => {
+      renderCard(newCard);
+      console.log(newCard);
+      addCardFormValidator.disableSubmitButton();
+      newCardPopup.close();
+      addNewCardForm.reset();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => newCardPopup.setLoading(false));
 }
 
-function handleEditSubmit(inputData) {
-  userInfo.setUserInfo({
-    title: inputData.profile__name,
-    description: inputData.profile__description,
-  });
-  profileEditPopup.close();
+function handleEditSubmit(inputValues) {
+  const { profile__name, profile__description } = inputValues;
+  profileEditPopup.setLoading(true);
+  api
+    .updateProfile({ name: profile__name, description: profile__description })
+    .then(() => {
+      userInfo.setUserInfo(
+        {
+          name: inputValues.profile__name,
+          description: inputValues.profile__description,
+        },
+        profileEditPopup.close()
+      );
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => profileEditPopup.setLoading(false));
 }
 
 function handleCardImageClick(name, link) {
   popupImage.open(name, link);
 }
 
-const cardDelete = document.querySelectorAll("#delete-image-close");
-//const cardDeleteModal = document.querySelectorAll("#delete-image-modal");
-
-function handleDeleteCard(card) {
+function handleDeleteCard(card, cardId) {
   modalWithConfirm.setSubmitFunction(() => {
     api
-      .deleteCard(card)
+      .deleteCard(cardId)
       .then(() => {
-        card.remove();
+        card._handleCardDelete();
+        modalWithConfirm.close();
       })
-      .catch((err) => console.error(err));
+      .catch((err) => {
+        console.error(err);
+      });
   });
   modalWithConfirm.open();
 }
 
+function handleEditPrfilePic(url) {
+  const { profile__url } = url;
+  profilePicPopup.setLoading(true);
+  api
+    .updateAvatar(profile__url)
+    .then((users) => {
+      userInfo.setUserAvatar(users.avatar), profilePicPopup.close();
```

Comment:
`      userInfo.setUserAvatar(users.avatar), profilePicPopup.close();
`

You don't need `,` here


```js
    .then((users) => {
      userInfo.setUserAvatar(users.avatar);
      profilePicPopup.close();
    })

```